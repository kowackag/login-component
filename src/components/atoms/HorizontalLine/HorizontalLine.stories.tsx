import type { Meta, StoryObj } from "@storybook/react";

import { HorizontalLine } from "./HorizontalLine";

const meta: Meta<typeof HorizontalLine> = {
	component: HorizontalLine,
      decorators: [
		(Story) => (
			<div style={{ display:"flex", width: "360px" }}>
				<Story />
			</div>
		),
	],
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
	args: {},
};
