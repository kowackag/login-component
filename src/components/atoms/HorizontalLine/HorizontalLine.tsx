export const HorizontalLine = () => {
	return <span data-testid="hrLine" className="flex-grow border border-b border-b-gray-300"></span>;
};
