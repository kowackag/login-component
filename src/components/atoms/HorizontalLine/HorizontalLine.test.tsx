import { expect, describe, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { HorizontalLine } from "./HorizontalLine";

describe("Horizontal Line ", () => {
	it("is visible", () => {
		render(<HorizontalLine />);
		expect(screen.getByTestId("hrLine")).toBeVisible();
	});
	it("renders correctly", () => {
		const wrapper = render(<HorizontalLine />);
		expect(wrapper.container).toMatchSnapshot();
	});
});
