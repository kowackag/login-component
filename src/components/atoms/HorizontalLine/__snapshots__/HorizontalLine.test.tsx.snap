// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Horizontal Line  > renders correctly 1`] = `
<div>
  <span
    class="flex-grow border border-b border-b-gray-300"
    data-testid="hrLine"
  />
</div>
`;
